src/Domain/CoordinateMaps/FocallyLiftedEndcap.cpp

```diff
@@ -0,0 +1,297 @@
+             ? -cube(a) / 3.0
+             : (a * cos(ax) - sin(ax) / x) / square(x);
+}
+double dlogx_sin_ax_over_x(double x, double a) noexcept {
+  return dlogx_sin_ax_over_x(x, a * x, a);
+}
+DataVector dlogx_sin_ax_over_x(const DataVector& x, double a) noexcept {
+  DataVector result(x);
+  for (size_t i = 0; i < x.size(); ++i) {
+    result[i] = dlogx_sin_ax_over_x(x[i], a * x[i], a);
+  }
+  return result;
+}
+}  // namespace Endcap_detail
+
+Endcap::Endcap(const std::array<double, 3>& center, double radius,
+               double z_plane) noexcept
+    : center_(center),
+      radius_([&]() noexcept {
+        // The equal_within_roundoff below has an implicit scale of 1,
+        // so the ASSERT may trigger in the case where we really
+        // want an entire domain that is very small.
+        ASSERT(not equal_within_roundoff(radius, 0.0),
+               "Cannot have zero radius");
+        return radius;
+      }()),
+      theta_(acos((z_plane - center_[2]) / radius_)) {
+  ASSERT(z_plane != center[2],
+         "Plane must intersect sphere at more than one point");
+}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 3> Endcap::operator()(
+    const std::array<T, 3>& source_coords) const noexcept {
+  using return_type = tt::remove_cvref_wrap_t<T>;
+  const return_type& xbar = source_coords[0];
+  const return_type& ybar = source_coords[1];
+  const return_type rho = sqrt(square(xbar) + square(ybar));
+  const return_type sin_factor =
+      radius_ * Endcap_detail::sin_ax_over_x(rho, theta_);
+  const return_type z = radius_ * cos(rho * theta_) + center_[2];
+  const return_type x = sin_factor * xbar + center_[0];
+  const return_type y = sin_factor * ybar + center_[1];
+  return std::array<return_type, 3>{{std::move(x), std::move(y), std::move(z)}};
+}
+
+template <typename T>
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> Endcap::jacobian(
+    const std::array<T, 3>& source_coords) const noexcept {
+  using return_type = tt::remove_cvref_wrap_t<T>;
+  const return_type& xbar = source_coords[0];
+  const return_type& ybar = source_coords[1];
+  const return_type rho = sqrt(square(xbar) + square(ybar));
+  const return_type sin_factor =
+      radius_ * Endcap_detail::sin_ax_over_x(rho, theta_);
+  const return_type d_sin_factor =
+      radius_ * Endcap_detail::dlogx_sin_ax_over_x(rho, theta_);
+
+  auto jacobian_matrix =
+      make_with_value<tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame>>(
+          dereference_wrapper(source_coords[0]), 0.0);
+
+  // dz/dxbar
+  get<2, 0>(jacobian_matrix) = -sin_factor * theta_ * xbar;
+  // dz/dybar
+  get<2, 1>(jacobian_matrix) = -sin_factor * theta_ * ybar;
+  // dx/dxbar
+  get<0, 0>(jacobian_matrix) = d_sin_factor * square(xbar) + sin_factor;
+  // dx/dybar
+  get<0, 1>(jacobian_matrix) = d_sin_factor * xbar * ybar;
+  // dy/dxbar
+  get<1, 0>(jacobian_matrix) = d_sin_factor * ybar * xbar;
```

Comment:
is it faster to avoid the repeat calculation by setting the 1,0 term to the 0,1 term?